data-prepper-plugins/dynamodb-source/src/main/java/org/opensearch/dataprepper/plugins/source/dynamodb/stream/StreamScheduler.java

```diff
@@ -15,25 +15,41 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.time.Duration;
 import java.util.Optional;
+import java.util.Random;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.function.BiConsumer;
 
+import static com.google.common.math.LongMath.pow;
+import static com.google.common.primitives.Longs.min;
+import static java.lang.Math.max;
+
 /**
  * A scheduler to manage all the stream related work in one place
  */
 public class StreamScheduler implements Runnable {
     private static final Logger LOG = LoggerFactory.getLogger(StreamScheduler.class);
+    private static final Logger SHARD_COUNT_LOGGER = LoggerFactory.getLogger("ShardCountLogger");
```

Comment:
This logger should still have a name in the `org.opensearch.dataprepper.plugins.source.dynamodb` package, or even in `org.opensearch.dataprepper.plugins.source.dynamodb.stream`.

Without this, you won't be able to turn on all DynamoDB logging by setting ` log.level.org.opensearch.dataprepper.plugins.source.dynamodb = TRACE`.